src/nxt_unit.c

```diff
@@ -3224,23 +3224,20 @@ nxt_unit_buf_read(nxt_unit_buf_t **b, uint64_t *len, void *dst, size_t size)
     while (buf != NULL) {
         last_buf = buf;
 
-        copy = buf->end - buf->free;
-        copy = nxt_min(rest, copy);
+        copy = nxt_min(rest, buf->end - buf->free);
```

Comment:
1. `buf` can be used for both request and response, the size of `end - free` means free size. Take a look at this.
```
#define nxt_buf_mem_used_size(bm)                                             \
    ((bm)->free - (bm)->pos)

#define nxt_buf_mem_free_size(bm)                                             \
    ((bm)->end - (bm)->free)
```
The logic here means how many bytes can be written.
```
/* It should be the free size of the buf. */
copy = buf->end - buf->free;

/* It should be no larger than the `rest` size. */
copy = nxt_min(rest, copy);
```
It's clear. And I'm afraid this doesn't improve the code readability.
```
copy = nxt_min(rest, buf->end - buf->free);
```
Since we need to understand what `buf->end - buf->free` means. Actually, it means free space to copy.
```
copy = buf->end - buf->free;
```
I would say both of them are fine, if yes, I suggest not touching it.

Reply:
Look at the next line:

```diff
-        copy = buf->end - buf->free;
-        copy = nxt_min(rest, copy);
+        copy = nxt_min(rest, buf->end - buf->free);
 
         p = nxt_cpymem(p, buf->free, copy);
```

We're copying _from_ `buf` into `p`.  We're _not_ copying into `buf`.